### ISSUE TITLE
Amazon S3

### ISSUE THREAD

**Issue by muleymadness** [kylegilman/video-embed-thumbnail-generator#177]
Does this plugin have the capability to work with Amazon S3, if so how and if not would this be something you could add?  My server storage size will fill up quickly with all of my video files.

**Comment by kylegilman** [comment]
Closing old issues. If you still want support for this, post a comment and I will re-open.

**Comment by muleymadness** [comment]
So I'm curious as if you'd look at this again.  I've created a S3/Cloudfront account and uploaded a video with your plugin.  The video appears to now be stored on S3 and is working great, video appears to have been deleted from wordpress media library (not stored locally).  And I am able to view, download it, etc. without issue...as near as I can tell.  It will not however generate a thumbnail.  Wondering if this is something you'd be willing to look into to see if you could get it to work? I'd love to have a thumbnail attached to each video that is also pulled/stored from Amazon S3 and not locally.  Or even if it's stored locally that would work also, but it won't create one at all now.

**Comment by muleymadness** [comment]
I suppose (which I tested and it does appear to work) I can upload a separate thumbnail, then paste this into the thumbnail url field then set as featured image.  This is of course a bit of a pain, and would be nice if the plugin did it naturally.

**Comment by kylegilman** [comment]
Do you have FFMPEG on your server? It should work if you have FFMPEG. If not, there is a way to set up a CORS policy on your Amazon bucket to allow the plugin to run the magic necessary to create thumbnails in the browser. I don't fully understand the CORS process but you basically have to allow requests from your domain using "AllowedOrigin" in your CORS policy document. http://docs.aws.amazon.com/AmazonS3/latest/dev/cors.html

**Comment by muleymadness** [comment]
Yes I do have FFMPEG installed on server, but it's not working still.  Worked great until I tried it with my plugin called 'Amazon S3 and CloudFront'.  Now it doesn't create thumbnails anymore.  Not sure what CORS is either, looks kinda complicated but I can take a look there.

**Comment by muleymadness** [comment]
If I check to only use FFMPEG and then check the box to generate a thumbnail automatically upon upload it DOES try to generate a thumbnail, only problem is the image is not correct ever (mostly just a grayed out thumbnail).  Never correctly pulls a thumbnail from the video but it does create a bad one.

Could it be my FFMPEG version is out of date?

**Comment by muleymadness** [comment]
Looks my version is 0.11.3, which appears to be really old.  Maybe that's the problem?

**Comment by kylegilman** [comment]
I don't know why you'd have a problem with remote thumbnails just because it's an older version, but FFMPEG is mysterious. Can you generate thumbnails if you enter the URL manually in the Embed Video From URL tab? Are you requiring HTTPS on S3? FFMPEG can't read the video stream if it's encrypted.

**Comment by muleymadness** [comment]
Nope on URL manually from URL tab, doesn't work either same result as other.  Nope don't believe I'm requiring HTTPS on S3 either.  It doesn't generate a thumbnail on upload, just grayed out and a messed up thumbnail.  If I try to regenerate thumbnails after the video is initially uploaded it does absolutely nothing and won't generate any.  Only on initial video upload and always messed up image.  Frustrating...not clue what to do.

**Comment by kylegilman** [comment]
Can you give me the URL of one of these videos that doesn't work?

**Comment by muleymadness** [comment]
Last video on page 2

http://www.muledeerphotography.com/video/turkey-trail-cam-clips/

**Comment by muleymadness** [comment]
I've deleted that video/link.  Sorry about the trouble, but I've decided against trying to get this to work with S3 & Cloudfront for the the time being.  It's to big of a headache, I've going back to my original method which was working for now.  I've deactivated the plugins that was using the above link, and it messed up the link so I deleted the video.

**Comment by muleymadness** [comment]
Last video on page 2 here...
http://www.muledeerphotography.com/video/turkey-trail-cam-clips/

**Comment by muleymadness** [comment]
Interestingly enough, I deactivated my S3 plugins (that I though was causing the problems) and went back to my original method with just using your plugin with wordpress and it doesn't work anymore like it use to.  Meaning I was getting the thumbnails working originally and now not.

**Comment by muleymadness** [comment]
I've uninstalled my S3 plugins(which was apparently causing it not to work) and it now works fine again, just to big of a headache, so I'm not going to use S3 or CloudFront for the time being.

**Comment by kylegilman** [comment]
You're using an unreleased development version of the plugin downloaded from GitHub. I can't guarantee that anything will work when using the development files, and I don't recommend it. I can't tell what version those files are, and there's always a chance you just happened to download it after I broke something. I tried the S3 and Cloudfront plugin and ran into a couple snags that I just fixed, but I can't duplicate the corrupt thumbnails you're getting. I used an even older version of FFMPEG on the video you were having trouble with and it worked fine. I'm stumped.
